agents/chopper_mc2000b/chopper_mc2000b_agent.py

```diff
@@ -0,0 +1,339 @@
+                                 buffer_time=1
+                                 )
+
+        self.agent.register_feed('output_freqs',
+                                 record=True,
+                                 agg_params=agg_params,
+                                 buffer_time=1
+                                 )
+
+    @ocs_agent.param('auto_acquire', default=False, type=bool)
+    def init_chopper(self, session, params):
+        """init_chopper(auto_acquire=False)
+
+        **Task** - Perform first time setup of MC2000B chopper controller
+        communication.
+
+        Parameters
+        __________
+
+        auto_acquire : bool, optional
+            Default is false. Starts data acquisition after
+            initilialization if True.
+        """
+        if self.initialized:
+            return True, "Already initialized"
+
+        with self.lock.acquire_timeout(job='init_chopper') as acquired:
+            if not acquired:
+                self.log.warn(f"Could not start Task because "
+                              f"{self.lock.job} is already running")
+                return False, "Could not acquire lock"
+
+            session.set_status('running')
+
+            # Establish connection to the chopper controller
+            self.hdl = MC2000BOpen(self.comport, self.nbaud, self.timeout)
+
+        if (self.hdl == 0):
+            self.initialized = True
+            self.log.info("Chopper connected")
+        else:
+            self.log.warn("Chopper not connected")
+
+        # Start data acquisition if requested
+        if params['auto_acquire']:
+            resp = self.agent.start('acq', params={})
+            self.log.info(f'Response from acq.start(): {resp[1]}')
+
+        return True, "Chopper controller agent initialized"
+
+    @ocs_agent.param('freq', type=int)
+    def set_frequency(self, session, params):
+        """set_frequency(freq=None)
+
+        **Task** - Set the frequency of the chopper.
+
+        Parameters
+        __________
+        freq : int
+            Frequency of chopper blades.
+        """
+        with self.lock.acquire_timeout(timeout=3, job='set_frequency') as acquired:
+            if not acquired:
+                self.log.warn(f"Could not start Task because "
+                              f"{self.lock.job} is already running")
+                return False, "Could not acquire lock"
+
+            session.set_status('running')
+
+            MC2000BSetFrequency(self.hdl, params['freq'])
+
+        return True, "Chopper frequency set to {} Hz".format(params['freq'])
+
+    @ocs_agent.param('bladetype', type=str, default='MC1F2')
+    def set_bladetype(self, session, params):
+        """set_bladetype(bladetype=None)
+
+        **Task** - Set the bladetype of the chopper. Selecting a bladetype
+        influences the range of frequencies permitted for the chopper.
+
+        Parameters
+        __________
+        bladetype : str
```

Comment:
Is there documentation from Thorlabs that we can link to that describes valid blade types? Or is this something that's obvious to users? If not, we should list valid blade types.